### PR TITLE
Port JS library from defunctzombie/node-dbmate

### PR DESCRIPTION
**Pull request by defunctzombie**
Add a library interface to the dbmate npm package. This interface allows developers to invoke `dbmate` from their node programs. A common usecase is to invoke dbmate commands during unit testing.

I tested this change by using `npm pack ./packages/dbmate` after running generate. This creates a `.tar.gz` file I can install using `yarn` or `npm`.

### CODE REVIEW

File: typescript/generate.ts [hunk 1]
```diff
@@ -113,6 +114,10 @@ async function main() {
   // copy readme and license
   await copyFile("../LICENSE", "packages/dbmate/LICENSE");
   await copyFile("../README.md", "packages/dbmate/README.md");
+
+  await exec("npm", ["run", "build"], {
+    cwd: "packages/dbmate",
+  });
```
Comment (defunctzombie) [inline typescript/generate.ts:120]: Another location for this could be the CI steps rather than here in `generate.ts`

**Comment by defunctzombie** [comment]
@amacneil PTAL

**Comment by amacneil** [comment]
Partially landed in https://github.com/amacneil/dbmate/pull/418

**Comment by defunctzombie** [comment]
Closing this out in favor of using `resolveBinary` from the published package and using our own wrapper for now.

**Pull request closed without merging**